unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs

```diff
@@ -55,37 +55,33 @@ public void Dispose()
             view.Dispose();
         }
 
-        DataStore.i.taskbar.isExploreV2Enabled.OnChange -= OnActivateFromTaskbar;
-
         if (placesAndEventsSectionController != null)
         {
             placesAndEventsSectionController.OnCloseExploreV2 -= OnCloseButtonPressed;
             placesAndEventsSectionController.Dispose();
         }
-
-        toggleExploreTrigger.OnTriggered -= OnToggleActionTriggered;
     }
 
-    public void SetVisibility(bool visible)
+    public void SetVisibility(bool visible) { isOpen.Set(visible); }
+
+    private void IsOpenChanged(bool current, bool previous) { SetVisibility_Internal(current); }
+
+    internal void SetVisibility_Internal(bool visible)
     {
         if (view == null)
             return;
 
-        if (visible != DataStore.i.exploreV2.isOpen.Get())
+        if (visible)
```

Comment:
I see that this is changing the logic. Are we 100% sure that this can't be called twice with a `true` value?

Reply:
The logic has been changed a bit further.

Now `SetVisibility` only modifies `isOpen` in the DataStore which triggers `IsOpenChanged()` (if the value has been actually changed) which ends up calling `SetVisibility_Internal` and performs the visibility change.

Basically I'm relying on the DataStore itself which ensures the `OnChanged` is not triggered if the value is not modified (unless you want to force the event trigger). In any case it should work properly.